plugin/completion/flags_manager.py

```diff
@@ -346,44 +380,47 @@ def flags_from_clang_file(file, separate_includes):
         flags = set()
         with open(file.full_path()) as f:
             content = f.readlines()
-            flags = FlagsManager.parse_flags(file.folder(),
-                                             content,
-                                             separate_includes)
+            flags = FlagsManager.parse_flags(file.folder(), content)
         log.debug(" .clang_complete contains flags: %s", flags)
         return flags
 
     @staticmethod
-    def parse_flags(folder, lines, separate_includes):
+    def parse_flags(folder, lines):
         """
         Parse the flags in a given file
 
         Args:
             folder (str): current folder
             lines (str[]): lines to parse
-            separate_includes (bool): if True "-I/blah" turns to "-I '/blah'"
 
         Returns:
             str[]: flags
         """
-        mask = '{}{}'
-        separate_mask = '{} "{}"'
+
+        def split_if_include(flag):
+            """ Split flag if it is an include flag
+
+            Args:
+                flag (str): flag to test and split if needed
+
+            Returns:
+                (str, str): tuple of (prefix, path)
+            """
+            for prefix in FlagsManager._include_prefixes:
+                if flag.startswith(prefix):
+                    return (prefix, flag[len(prefix):])
+            return (None, None)
+
         flags = set()
-        log.debug(" all lines: %s", lines)
         for line in lines:
-            for prefix in FlagsManager._possible_prefixes:
-                full_prefix = '-' + prefix
-                if not line.startswith('-'):
-                    line = '-' + line
-                if line.startswith(full_prefix):
-                    flag_content = line[len(full_prefix):].strip()
-                    flag_content = flag_content.strip('"')
-                    if prefix in FlagsManager._include_prefixes:
-                        if not path.isabs(flag_content):
-                            flag_content = path.join(folder, flag_content)
-                        if separate_includes:
-                            flags.add(separate_mask.format(full_prefix, flag_content))
-                        else:
-                            flags.add(mask.format(full_prefix, flag_content))
-                    else:
-                        flags.add(mask.format(full_prefix, flag_content))
+            line = line.strip()
+            if not line.startswith('-'):
+                line = '-' + line
```

Comment:
More precisely, clang++ accepts same arguments on any platform. Only clang-cl additionally supports Windows CL-style arguments. But in any case, it would be wrong to add dash on any platform. Unless the intention was to correct user's mistakes if he forgotten dash.